### DISCUSSION TITLE
How to update multiple rows ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [vitaly-t/pg-promise#915] [Q&A] [answered]
Hello,

how can I update multiple rows ?

this is my current code

```
    const cs = new pgp.helpers.ColumnSet(
      ['user_id', 'tenant_id', 'location_id', 'start_time', 'end_time', 'time_range', 'status', 'info'], { table: 'workers_plan' });

    const cs_values: UpdateWorkersPlan[] = plan.map((el: UpdateWorkersPlan) => ({
      plan_id: el.plan_id,
      user_id: el.user_id,
      tenant_id,
      location_id: el.location_id,
      start_time: el.start_time,
      end_time: el.end_time,
      time_range: tsrange(el.start_time, el.end_time),
      status: el.status,
      info: el.info,
    }));

    pgp.as.format('WHERE plan_id = $1', [how to access plan id ?])

    await pgp.helpers.update(cs_values, cs)
```

in pgp.as.format I have no plan id only when I map but I think I should not map all my pgp.as.format or how can I access it ?

**Comment by vitaly-t** [comment, marked as answer]
Again, see [Column](https://vitaly-t.github.io/pg-promise/helpers.Column.html), which supports `cnd` flag, so you can have columns that are just for setting the condition. There are plenty of examples of it on StackOverflow. [Here's](https://stackoverflow.com/questions/68451588/pg-promise-update-multiple-not-setting-columns-correctly) just one example.

You would simply have one column name as `?plan_id` or `{name: 'plan_id', cnd: true}`, and use it with `WHERE` clause.

See [helpers.update](https://vitaly-t.github.io/pg-promise/helpers.html#.update).

**Reply by pullmann4rent** [reply]
Thank you very much! but I got this error:

`error: column "location_id" is of type integer but expression is of type text`

My location_id is INT NULL so it can be a integer or null, in my body I send location_id as null but I got this error above. How can I fix it ?
And if I remove location_id then I got error to the next column:

`error: column "start_time" is of type timestamp without time zone but expression is of type text`

**Reply by vitaly-t** [reply]
So apply type casting - attribute `cast` in the `ColumnSet`, according to expected types.

**Reply by pullmann4rent** [reply]
I did this:

```
         {
          name: 'start_time',
          mod: ':raw',
          init(c) {
              return pgp.as.format("to_timestamp($1)", c.value);
          }
        },
```
But I got error 

`error: invalid input syntax for type double precision: "2026-10-28 20:00:00"`

**Reply by vitaly-t** [reply]
I suggested `cast` attribute, and you did something else entirely, now asking why it doesn't work :)

**Reply by vitaly-t** [reply]
What I suggested earlier, was `{name: 'start_time', cast: 'timestamptz'}`.

**Comment by vitaly-t** [comment]
I saw [your question on StackOverflow](https://stackoverflow.com/questions/78097197/how-to-update-multiple-rows-with-pg-promise), and I answered it.
